app/src/main/java/com/ubergeek42/WeechatAndroid/service/Events.java

```diff
@@ -77,4 +80,54 @@ public static void fireInput(@NonNull Buffer buffer, @Nullable String input) {
             return "SendMessageEvent(message=" + message + ")";
         }
     }
+
+    // https://weechat.org/files/doc/devel/weechat_relay_protocol.en.html#commands
+    // make use of the message identifier so we can correlate the server's response to our message
+    public static class SendMessageWithPromiseEvent {
+        final public String message;
+        final public CompletableFuture<RelayObject[]> future;
+        final public Integer futureId;
+        // The (id) field of the message is also used in BufferList.java for message handlers
+        // e.g. (listbuffers), (nicklist). We may want to add similar handlers for messages
+        // of this type, hence originalMessageId may be used here to indicate the original "purpose"
+        // of the message (see onMessage of RelayService.java)
+        final public String originalMessageId;
+
+        private static int futureCounter = 0;
+        // assumption that there won't be more than 1000 pending callbacks to be stored in
+        // the promiseCallbacks HashMap in RelayService.java at any instant;
+        // our message ID goes from 0 to 999
+        private static synchronized int incrementFutureCounter() {
+            return futureCounter++ % 1000;
+        }
+
+        private SendMessageWithPromiseEvent(String originalMessageId, String message,
+                                            CompletableFuture<RelayObject[]> future) {
+            this.originalMessageId = originalMessageId;
+            this.future = future;
+            this.futureId = incrementFutureCounter();
+            this.message = String.format("(%s) %s", String.valueOf(this.futureId), message);
```

Comment:
AS says "Unnecessary 'String.valueOf()' call"

Reply:
Reverted to the original `Events.java` before my PR since I'm reusing the existing counter in `BufferList.java` instead of rolling my own now.